frontend/src/components/pages/TranslationPage.tsx

```diff
@@ -73,14 +76,35 @@ const TranslationPage = () => {
     }
   };
 
+  const updatePercentageComplete = (
+    translatedLines: number,
+    storyLines: number,
+  ) => {
+    setPercentageComplete((translatedLines / storyLines) * 100);
+    setNumTranslatedLines(translatedLines);
+  };
+
   const onChangeTranslationContent = async (
     newContent: string,
     lineIndex: number,
   ) => {
     const updatedContentArray = [...translatedStoryLines];
     const index = arrayIndex(lineIndex);
-    updatedContentArray[index].translatedContent = newContent;
 
+    if (
+      // user deleted translation line
+      !newContent.trim() &&
+      translatedStoryLines[index]?.translatedContent?.trim()
```

Comment:
Hmm why a null check on `translatedStoryLines` and `translatedContent`? The only time I would expect .translatedContent to be null is on the super initial render, but it should already exist here

Reply:
I removed the null check on `translatedStoryLines` but get an "Object is possibly undefined" error without the optional chaining for `translatedContent`

Reply:
Yeah I'd argue that it should be `translatedContent!!` since something is very wrong if it's null? Don't know if it's bad practice but I'd rather it avoid "absorbing" errors